dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/UsersServiceImpl.java

```diff
@@ -206,9 +206,41 @@ public User createUser(String userName,
         User user = new User();
         Date now = new Date();
 
-        user.setUserName(userName);
-        user.setUserPassword(EncryptionUtils.getMd5(userPassword));
-        user.setEmail(email);
+        if (StringUtils.isNotEmpty(userName)) {
+
+            if (!CheckUtils.checkUserName(userName)) {
+                throw new org.apache.dolphinscheduler.service.exceptions.ServiceException(String.format("userName %s doesn't valid", userName));
+            }
+
+            User tempUser = userMapper.queryByUserNameAccurately(userName);
+            if (tempUser != null) {
+                throw new org.apache.dolphinscheduler.service.exceptions.ServiceException(String.format("userName %s exist", userName));
+            }
+            user.setUserName(userName);
+        }
+
+        if (StringUtils.isNotEmpty(userPassword)) {
```

Comment:
Please merge if statement

Reply:
> Please merge if statement

Sorry, I don't get your point...

Reply:
I mean please fix code smell

Reply:
I think OK, and merging can lead to unclear exception semantics.
sometimes, code smell needs to be judged based on the actual situation.

Reply:
Yeah, agree with Calvin